src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerExtensions.Obsolete.cs

```diff
@@ -0,0 +1,126 @@
+            return factory;
+        }
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [Obsolete("This method is retained only for compatibility. The recommended alternative is AddConsole(this ILoggingBuilder builder).", error: true)]
+        public static Logging.ILoggerFactory AddConsole(this Logging.ILoggerFactory factory, System.Func<string, Logging.LogLevel, bool> filter)
+        {
+            factory.AddConsole(filter, includeScopes: false);
+            return factory;
+        }
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [Obsolete("This method is retained only for compatibility. The recommended alternative is AddConsole(this ILoggingBuilder builder).", error: true)]
+        public static Logging.ILoggerFactory AddConsole(this Logging.ILoggerFactory factory)
+        {
+            return factory.AddConsole(includeScopes: false);
+        }
+
+        [Obsolete]
+        private sealed class ConsoleLoggerSettingsAdapter : IConfigureOptions<ConsoleLoggerOptions>, IOptionsChangeTokenSource<ConsoleLoggerOptions>
+        {
+            private IConsoleLoggerSettings _settings;
+            private ConsoleLoggerSettingsAdapter(IConsoleLoggerSettings settings)
+            {
+                _settings = settings;
+            }
+
+            IChangeToken IOptionsChangeTokenSource<ConsoleLoggerOptions>.GetChangeToken() => _settings.ChangeToken ?? NullChangeToken.Instance;
+
+            string IOptionsChangeTokenSource<ConsoleLoggerOptions>.Name => Microsoft.Extensions.Options.Options.DefaultName;
+
+            void IConfigureOptions<ConsoleLoggerOptions>.Configure(ConsoleLoggerOptions options)
+            {
+                options.IncludeScopes = _settings.IncludeScopes;
+                if (_settings is ConfigurationConsoleLoggerSettings configSettings)
+                {
+                    options.Configure(configSettings._configuration);
+                }
+                else if (_settings is ConsoleLoggerSettings consoleSettings)
+                {
+                    options.DisableColors = consoleSettings.DisableColors;
+                }
+            }
+
+            internal static OptionsMonitor<ConsoleLoggerOptions> GetOptionsMonitor(IConsoleLoggerSettings settings)
+            {
+                ConsoleLoggerSettingsAdapter adapter = new(settings);
+                OptionsFactory<ConsoleLoggerOptions> factory = new( new IConfigureOptions<ConsoleLoggerOptions>[] { adapter }, Array.Empty<IPostConfigureOptions<ConsoleLoggerOptions>>());
+                IOptionsChangeTokenSource<ConsoleLoggerOptions>[] sources = new IOptionsChangeTokenSource<ConsoleLoggerOptions>[] { adapter };
+                OptionsCache<ConsoleLoggerOptions> cache = new();
+
+                return new OptionsMonitor<ConsoleLoggerOptions>(factory, sources, cache);
+            }
+        }
+
+        private sealed class NullChangeToken : IChangeToken, IDisposable
```

Comment:
>NullChangeToken [](http://example.com/codeflow?start=29&length=15)

do we need to mark this as obsolete too? I am seeign we did that with `ConsoleLoggerSettingsAdapter`

Reply:
No, it's private.  I marked `ConsoleLoggerSettingsAdapter` obsolete because it needed to use other obsolete API.  This type does not, so it doesn't need obsolete.